src/fights.ts

```diff
@@ -323,157 +319,137 @@ function bestWitchessPiece() {
-              retrieveItem($item`pulled green taffy`)
-            ) {
-              setLocation($location`The Briny Deeps`);
-              meatOutfit(true, Requirement.merge(nextFight.requirements), true);
-              if (get("questS01OldGuy") === "unstarted") {
-                visitUrl("place.php?whichplace=sea_oldman&action=oldman_oldman");
-              }
-              if (!have($effect`Fishy`)) use($item`fishy pipe`);
-              nextFight.run({ location: $location`The Briny Deeps` });
-            } else if (nextFight.draggable) {
-              const type =
-                nextFight.name === "Backup" ? draggableFight.BACKUP : draggableFight.WANDERER;
-              const location = determineDraggableZoneAndEnsureAccess(type);
-              setLocation(location);
-              meatOutfit(true, Requirement.merge(nextFight.requirements));
-              nextFight.run({ location });
+            const romanticFamiliar = $familiars`Obtuse Angel, Reanimated Reanimator`.find(have);
+            if (romanticFamiliar && get("_badlyRomanticArrows") === 0 && nextFight.draggable) {
+              useFamiliar(romanticFamiliar);
             } else {
-              setLocation($location`Noob Cave`);
-              meatOutfit(true, Requirement.merge(nextFight.requirements));
-              nextFight.run({ location: $location`Noob Cave` });
+              meatFamiliar();
             }
+
+            const location = nextFight.draggable
+              ? checkUnderwater()
+                ? $location`The Briny Deeps`
+                : determineDraggableZoneAndEnsureAccess(nextFight.draggable)
+              : $location`Noob Cave`;
+
+            setLocation(location);
+            meatOutfit(true, Requirement.merge(nextFight.requirements));
+            nextFight.run({ location });
             postCombatActions();
           }
         });
 
         if (
           totalTurnsPlayed() - startTurns === 1 &&
           get("lastCopyableMonster") === $monster`Knob Goblin Embezzler` &&
-          (nextFight.name === "Backup" ||
-            nextFight.name === "Powerful Glove" ||
-            nextFight.name === "Macrometeorite" ||
-            get("lastEncounter") === "Knob Goblin Embezzler")
+          (nextFight.wrongEncounterName || get("lastEncounter") === "Knob Goblin Embezzler")
         ) {
           embezzlerLog.initialEmbezzlersFought++;
         }
 
         nextFight = getNextEmbezzlerFight();
 
+        // try to deliver the thesis
         const romanticMonsterPossible =
           (getCounter("Romantic Monster Window end") === -1 &&
             getCounters("Romantic Monster Window end", -1, -1).trim() === "") ||
           getCounter("Romantic Monster Window begin") > 0;
-        if (
-          !(nextFight && ["Backup", "Digitize", "Enamorang"].includes(nextFight.name)) &&
-          romanticMonsterPossible
-        ) {
+        if (!(nextFight && nextFight.draggable) && romanticMonsterPossible) {
```

Comment:
this logic is not correct. It should be 
```suggestion
        if (!romanticMonsterPossible && (!nextFight || !nextFight.draggable)) {
```
You can de morgan from there, which could bring it to something like
`!romanticMonsterPossible && !(nextFight && nextFight.draggable)`
which could then be
`!(romanticMonsterPossible || (nextFight && nextFight.draggable)`
But definitely it's wrong as is.

Reply:
@pstalcup